iothub/service/src/Jobs/DeviceJobParameters.cs

```diff
@@ -27,23 +25,12 @@ public DeviceJobParameters(JobType jobType, string deviceId)
         public DeviceJobParameters(JobType jobType, IEnumerable<string> deviceIds)
             : base(jobType)
         {
-            IList<string> deviceList = deviceIds?.ToListSlim();
```

Comment:
We can let the service validate the input, and no longer need to convert IEnumerable to IList.